### PR TITLE
b2sdk: 2.5.1 -> 2.6.0

### PR DESCRIPTION
**Pull request by philipmw**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] (on x86_64 only) Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by Stunkymonkey** [comment]
Result of `nixpkgs-review pr 355141` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>10 packages built:</summary>
  <ul>
    <li>backblaze-b2</li>
    <li>backblaze-b2.dist</li>
    <li>deja-dup</li>
    <li>duplicity</li>
    <li>duplicity.dist</li>
    <li>duply</li>
    <li>python311Packages.b2sdk</li>
    <li>python311Packages.b2sdk.dist</li>
    <li>python312Packages.b2sdk</li>
    <li>python312Packages.b2sdk.dist</li>
  </ul>
</details>

**Review by Stunkymonkey — APPROVED** [review]

File: pkgs/development/python-modules/b2sdk/default.nix [hunk 1]
```diff
@@ -81,6 +80,6 @@ buildPythonPackage rec {
     homepage = "https://github.com/Backblaze/b2-sdk-python";
     changelog = "https://github.com/Backblaze/b2-sdk-python/blob/v${version}/CHANGELOG.md";
     license = licenses.mit;
-    maintainers = [ ];
+    maintainers = with maintainers; [ pmw ];
```
Comment (khaneliman) [inline pkgs/development/python-modules/b2sdk/default.nix:83]: Can you put this change in a separate commit, please.

**Comment by philipmw** [comment]
Done

File: pkgs/development/python-modules/b2sdk/default.nix (hunk 1 above)
Comment (philipmw) [reply pkgs/development/python-modules/b2sdk/default.nix]: Done

Comment (khaneliman) [reply pkgs/development/python-modules/b2sdk/default.nix]: Looks like it was just removed, you can re-add it as another commit "b2sdk: add maintainer pmw".

Comment (philipmw) [reply pkgs/development/python-modules/b2sdk/default.nix:83]: Done

**Review by khaneliman — COMMENTED** [review]
Just realized hash didn't get changed. That means it would say 2.6.0 but would still pull cached 2.5.1 version.

**Comment by philipmw** [comment]
Whoops, sorry about that. I will fix.

This is something I don't yet understand about Nixpkgs - why the version number seems disconnected from what's actually downloaded.

**Comment by khaneliman** [comment]
> Whoops, sorry about that. I will fix.
> 
> This is something I don't yet understand about Nixpkgs - why the version number seems disconnected from what's actually downloaded.

For a sanity check, you could add:

```nix
  nativeInstallCheckInputs = [ versionCheckHook ];
  doInstallCheck = true;
```

**Comment by philipmw** [comment]
>   nativeInstallCheckInputs = [ versionCheckHook ];

I tried this, but it seemed to have no effect. So I didn't add it.

I tested the latest derivation both on Mac M1 and an Intel x86_64, rebuilding *backblaze-b2* software on top of this change and using it from `./result` to sync a directory to Backblaze. The SDK package is working as expected.

**Comment by khaneliman** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 355141`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 10 packages built:</summary>
  <ul>
    <li>backblaze-b2</li>
    <li>backblaze-b2.dist</li>
    <li>deja-dup</li>
    <li>duplicity</li>
    <li>duplicity.dist</li>
    <li>duply</li>
    <li>python311Packages.b2sdk</li>
    <li>python311Packages.b2sdk.dist</li>
    <li>python312Packages.b2sdk</li>
    <li>python312Packages.b2sdk.dist</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 10 packages built:</summary>
  <ul>
    <li>backblaze-b2</li>
    <li>backblaze-b2.dist</li>
    <li>deja-dup</li>
    <li>duplicity</li>
    <li>duplicity.dist</li>
    <li>duply</li>
    <li>python311Packages.b2sdk</li>
    <li>python311Packages.b2sdk.dist</li>
    <li>python312Packages.b2sdk</li>
    <li>python312Packages.b2sdk.dist</li>
  </ul>
</details>

---
### `x86_64-darwin`
<details>
  <summary>:x: 3 packages failed to build:</summary>
  <ul>
    <li>duplicity</li>
    <li>duplicity.dist</li>
    <li>duply</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 6 packages built:</summary>
  <ul>
    <li>backblaze-b2</li>
    <li>backblaze-b2.dist</li>
    <li>python311Packages.b2sdk</li>
    <li>python311Packages.b2sdk.dist</li>
    <li>python312Packages.b2sdk</li>
    <li>python312Packages.b2sdk.dist</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:x: 3 packages failed to build:</summary>
  <ul>
    <li>duplicity</li>
    <li>duplicity.dist</li>
    <li>duply</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 6 packages built:</summary>
  <ul>
    <li>backblaze-b2</li>
    <li>backblaze-b2.dist</li>
    <li>python311Packages.b2sdk</li>
    <li>python311Packages.b2sdk.dist</li>
    <li>python312Packages.b2sdk</li>
    <li>python312Packages.b2sdk.dist</li>
  </ul>
</details>

**Comment by khaneliman** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 355141`

---
### `x86_64-darwin`
<details>
  <summary>:x: 3 packages failed to build:</summary>
  <ul>
    <li>duplicity</li>
    <li>duplicity.dist</li>
    <li>duply</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 6 packages built:</summary>
  <ul>
    <li>backblaze-b2</li>
    <li>backblaze-b2.dist</li>
    <li>python311Packages.b2sdk</li>
    <li>python311Packages.b2sdk.dist</li>
    <li>python312Packages.b2sdk</li>
    <li>python312Packages.b2sdk.dist</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:white_check_mark: 9 packages built:</summary>
  <ul>
    <li>backblaze-b2</li>
    <li>backblaze-b2.dist</li>
    <li>duplicity</li>
    <li>duplicity.dist</li>
    <li>duply</li>
    <li>python311Packages.b2sdk</li>
    <li>python311Packages.b2sdk.dist</li>
    <li>python312Packages.b2sdk</li>
    <li>python312Packages.b2sdk.dist</li>
  </ul>
</details>

**Review by khaneliman — APPROVED** [review]
Current duplicity failure exists on master for x86 darwin, but looks different. LGTM aarch64 might be transient because of the check phase length.
